From a22cd17534003e7148ef76c6f6541ae5d1a7c7c1 Mon Sep 17 00:00:00 2001
From: Vince Chang <vince_chang@aspeedtech.com>
Date: Wed, 16 Apr 2025 16:37:08 +0800
Subject: [PATCH] resolve segmentation fault in i2cdump

Fixes: 6b3686d12349 ("meta-phosphor: i2c-tools: rebase patch for 4.4")

Corrected the argument index for checking "d" from `argv[optind+3]` to
`argv[optind+2]`. The incorrect index caused segmentation fault.

Tested:
```
root@ast2700-default:~# i2cdump -fy 0 0x30 i
No segmentation fault
```

Upstream-Status: Pending

Signed-off-by: Vince Chang <vince_chang@aspeedtech.com>
---
 tools/i2cdump.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/i2cdump.c b/tools/i2cdump.c
index 3a55a95..ad7f19d 100644
--- a/tools/i2cdump.c
+++ b/tools/i2cdump.c
@@ -175,9 +175,9 @@ int main(int argc, char *argv[])
 	} else if (!strncmp(argv[optind+2], "c", 1)) {
 		size = I2C_SMBUS_BYTE;
 		pec = argv[optind+2][1] == 'p';
-	} else if (!strncmp(argv[optind+3], "d", 1)) {
+	} else if (!strncmp(argv[optind+2], "d", 1)) {
 		size = I2C_SMBUS_DWORD;
-		pec = argv[optind+3][1] == 'p';
+		pec = argv[optind+2][1] == 'p';
 	} else if (!strcmp(argv[optind+2], "i"))
 		size = I2C_SMBUS_I2C_BLOCK_DATA;
 	else {
-- 
2.43.0

